extensions/MultiUpload/multiupload.js

```diff
@@ -264,12 +264,12 @@ function fillDestFilename( id ) {
 	// URLs are less likely to have a useful extension, so don't include them in the
 	// extension check.
 	current = id.replace( /[^0-9]/g, '' );
-	if( wgFileExtensions && id != 'wpUploadFileURL' ) {
+	if( wgFileExtensions && id !== 'wpUploadFileURL' ) {
 		var found = false;
-		if( fname.lastIndexOf( '.' ) != -1 ) {
-			var ext = fname.substr( fname.lastIndexOf( '.' ) + 1 );
-			for( var i = 0; i < wgFileExtensions.length; i++ ) {
-				if( wgFileExtensions[i].toLowerCase() == ext.toLowerCase() ) {
+		if ( fname.lastIndexOf('.') !== -1 ) {
+			var ext = fname.substr( fname.lastIndexOf('.') + 1 ).toLowerCase();
+			for (var wgfi in wgFileExtensions) {
```

Comment:
[The upstream version of MultiUpload](https://github.com/wikimedia/mediawiki-extensions-MultiUpload/blob/bfbfe7c6787b9201704e7abe6807031bc8cd076d/resources/upload.js.patched#L391) assumes it's an array. Interesting, did we change the way `wgFileExtensions` is exposed to JavaScript?